### ISSUE TITLE
[GV] Implement going back from YouTube with hardware back

### ISSUE THREAD

**Issue by dnarcese** [mozilla-mobile/firefox-tv#1837]
Currently, we use `WebView#copyBackForwardList()` in the implementation of exiting YouTube with hardware back (#1542). afaict (although I haven't investigated much), GV doesn't have this functionality. We can probably find a better way to implement this without needing the web history list.

**Comment by dnarcese** [comment]
Looks like we will still need to access the web history list. This is blocked until the functionality is added to GV and a-c:
https://bugzilla.mozilla.org/show_bug.cgi?id=1509110
https://github.com/mozilla-mobile/android-components/issues/1340

**Comment by cpeterso** [comment]
The GV API ([bug 1509110](https://bugzilla.mozilla.org/show_bug.cgi?id=1509110)) has not landed yet, but A-C might be able to expose WebView's copyBackForwardList() API in mozilla-mobile/android-components#1340 and then FFTV can develop against (`copyBackForwardList` issue #1558) while we wait for the GV API.

**Comment by psymoon** [comment]
@dnarcese Would this also simplify things in system WebView implementation as well? Since a-c will expose this through `EngineView`?

**Comment by dnarcese** [comment]
@psymoon Yes, once a-c exposes this API, we can get rid of our custom EngineView extensions for system WebView as well.

**Comment by devinreams** [comment]
Waiting on GV (in progress) and then will need to get put into A-C so maybe a sprint away..
